alacritty_terminal/src/term/mod.rs

```diff
@@ -1046,10 +1046,13 @@ impl<T> Term<T> {
         let c = self.grid.cursor.charsets[self.active_charset].map(c);
         let fg = self.grid.cursor.template.fg;
         let bg = self.grid.cursor.template.bg;
-        let flags = self.grid.cursor.template.flags;
+        let mut flags = self.grid.cursor.template.flags;
 
         let mut cursor_cell = self.grid.cursor_cell();
 
+        // Preserve PROMPT_MARK if we had it.
```

Comment:
```suggestion
        // Preserve PROMPT_MARK.
```

Also is this necessary? The prompt shouldn't usually be overwritten?

Reply:
It does get overwritten due to the way everyone is sending prompt marks. They send prompt mark and then start writing the prompt, leading to overwriting the cell. The documentation and recommendations state that you should send it before writing the prompt.

Reply:
Really not a big fan of this aspect of the escape. Either way though this comment should be fixed.